lib/uniffi-zcash-cli/src/diff/git.rs

```diff
@@ -0,0 +1,50 @@
+use anyhow::Context;
+use git2::Repository;
+
+const LIBS_REPO_URL: &str = "https://github.com/zcash/librustzcash.git";
+
+// The way the git tag is constructed, assumes that the git tags
+// will always be in the format "lib-semver"
+//
+// NOTE: The tag is constructed in a format used by `librustzcash`, which is "libName-livVersion".
+// For example: zcash_primitives-0.10.2
+// So if the naming convention is not applied, the code will not be able to checkout the tag.
+pub(crate) fn init_libs_repo(
+    lib_name: &String,
+    repo_path: String,
+    version: String,
+) -> anyhow::Result<()> {
+    let tag = format!("{}-{}", lib_name, version);
+    let repo = match Repository::open(repo_path.to_owned()) {
+        Ok(r) => r,
+        Err(_) => {
+            Repository::clone(LIBS_REPO_URL, repo_path).context("failed to clone current repo")?
+        }
+    };
+
+    if repo.find_reference(&tag).is_err() {
```

Comment:
shouldn't this be reversed? If you can't find a ref then how do you want to checkout it

Reply:
I think I added it, while there was a possibility for a conflict if the repo exists on the user's machine.
Since I added the `tempdir` change, I removed the if statement completely and it still works as intended.